### PR TITLE
fix(footer): fix for store initialization

### PR DESCRIPTION
**Pull request by jeffchew**
### Related Ticket(s)

Refs #3589, #3590, #3594 

### Description

This PR was created off of PR #3590 in order to fix the store initialization, but also updating the snapshot files. In addition, there are some knob changes to include the ability to switch footer styles in the dotcomshell, and also removing knobs we do not with to expose to adopters.

There was also an issue with the footer links not getting mapped properly to the footer in the dotcomshell (default footer), which is fixed here as well.

### Changelog

**Changed**

- Updated snapshot files
- Updated knobs in the dotcom-shell
- Fix to footer (default) links loading in dotcomshell

### CODE REVIEW

File: packages/web-components/src/components/dotcom-shell/__stories__/dotcom-shell.stories.ts [hunk 1]
```diff
@@ -109,21 +108,23 @@ export default {
   parameters: {
     ...readme.parameters,
     knobs: {
-      'dds-masthead-container': ({ groupId }) => ({
+      Masthead: ({ groupId }) => ({
```
Comment (asudoh) [inline packages/web-components/src/components/dotcom-shell/__stories__/dotcom-shell.stories.ts:111]: Just curious: Is it your preference of knob namespacing?

File: packages/web-components/src/components/footer/footer-container.ts [hunk 2]
```diff
@@ -107,6 +107,7 @@ export function mapStateToProps(state: FooterContainerState): FooterContainerSta
     langDisplay,
     localeList: !language ? undefined : localeLists?.[language],
     links: !language ? undefined : translations?.[language]?.footerMenu,
+    footerLinks: !language ? undefined : translations?.[language]?.footerMenu,
```
Comment (asudoh) [inline packages/web-components/src/components/footer/footer-container.ts:110]: We need to fix this in `<dds-dotcom-shell-container>`, at: https://github.com/carbon-design-system/ibm-dotcom-library/blob/release/v1.10.0/packages/web-components/src/components/dotcom-shell/dotcom-shell-container.ts#L46

File: packages/web-components/src/components/dotcom-shell/__stories__/dotcom-shell.stories.ts (hunk 1 above)
Comment (jeffchew) [reply packages/web-components/src/components/dotcom-shell/__stories__/dotcom-shell.stories.ts:111]: Yes, namely for labeling the tabs in the knob panel so it's clearer to the adopters.

File: packages/web-components/src/components/footer/footer-container.ts (hunk 2 above)
Comment (jeffchew) [reply packages/web-components/src/components/footer/footer-container.ts:110]: Hmm ok, perhaps a follow-up PR after merging?

File: packages/web-components/src/components/dotcom-shell/__stories__/dotcom-shell.stories.ts [hunk 3]
```diff
@@ -80,9 +80,9 @@ const StoryContent = () => html`
 `;
 
 export const Default = ({ parameters }) => {
-  const { brandName, userStatus, navLinks } = parameters?.props?.['dds-masthead-container'] ?? {};
+  const { brandName, userStatus, navLinks } = parameters?.props?.Masthead ?? {};
   const { langDisplay, language, size: footerSize, legalLinks, links: footerLinks, localeList } =
-    parameters?.props?.['dds-footer-composite'] ?? {};
+    parameters?.props?.Footer ?? {};
```
Comment (asudoh) [inline packages/web-components/src/components/dotcom-shell/__stories__/dotcom-shell.stories.ts:85]: For fixing Prettier error:

```suggestion
```

File: packages/web-components/src/components/dotcom-shell/__stories__/dotcom-shell.stories.ts [hunk 4]
```diff
@@ -80,9 +80,9 @@ const StoryContent = () => html`
 `;
 
 export const Default = ({ parameters }) => {
-  const { brandName, userStatus, navLinks } = parameters?.props?.['dds-masthead-container'] ?? {};
+  const { brandName, userStatus, navLinks } = parameters?.props?.Masthead ?? {};
   const { langDisplay, language, size: footerSize, legalLinks, links: footerLinks, localeList } =
```
Comment (asudoh) [inline packages/web-components/src/components/dotcom-shell/__stories__/dotcom-shell.stories.ts:84]: For fixing Prettier error:

```suggestion
  const { langDisplay, language, size: footerSize, legalLinks, links: footerLinks, localeList } = parameters?.props?.Footer ?? {};
```

**Pull request merged**